### ISSUE TITLE
Two functions with one path produce inconsistent results

### ISSUE THREAD

**Issue by bbeattie-phxlabs** [tiangolo/fastapi#5543]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()


@api.get("/foo")
async def foo1() -> dict:
    """Foo #1"""
    return {"content": "First foo"}


@api.get("/foo")
async def foo2() -> dict:
    """Foo #2"""
    return {"content": "Second foo"}
```

### Description

* `GET /openapi.json` (and consequently Swagger/Redoc) refer to `/foo` with `Foo #2` documentation.
* `GET /foo` returns `First foo`.
* Expected `GET /foo` to return `Second foo` _or_ `GET /openapi.json` to return `Foo #1` documentation. As it stands, the results of overriding a path are inconsistent.

### Operating System

Linux

### Operating System Details

Debian 11.5

### FastAPI Version

0.85.1

### Python Version

3.10.8

### Additional Context

As for prior art, searching wider brings up https://stackoverflow.com/questions/72219039/how-to-overwrite-extend-fastapi-apirouter-route, though this doesn't address the inconsistency found.

**Comment by bbeattie-phxlabs** [comment]
https://github.com/tiangolo/fastapi/blob/master/fastapi/routing.py#L593 talks of `routes.append`. I _suspect_ this is an upstream Starlette issue. Digging in the code now...

**Comment by bbeattie-phxlabs** [comment]
As for prior art, searching wider brings up https://stackoverflow.com/questions/72219039/how-to-overwrite-extend-fastapi-apirouter-route, though this doesn't address the inconsistency found.

**Comment by odiseo0** [comment]
This is probably because the way FastAPI adds the routes. It should always take in count the first route.

**Comment by bbeattie-phxlabs** [comment]
Agreed that https://github.com/tiangolo/fastapi/blob/master/fastapi/routing.py#L593 is the route appending logic. My concern is more the inconsistency between the openapi.json content and the function the endpoint actually calls isn't the same.

**Comment by odiseo0** [comment]
Indeed, it's probably a bug. See https://fastapi.tiangolo.com/tutorial/path-params/?h=order#order-matters:~:text=Similarly%2C%20you%20cannot%20redefine%20a%20path%20operation%3A

He doesn't mention nothing about the docs, but it's probably that the logic isn't applied when the docs are being generated

**Comment by tiangolo** [comment]
Hmm, yeah that looks like a bug in the example code you showed, one thing would override the other, you are declaring the same path operationb twice.

**Comment by bbeattie-phxlabs** [comment]
Yep, I get my code is overriding the same path. I would hope that FastAPI would either disallow overriding (raise an exception?) or provide consistent behaviour (either's functionality and documentation). But what we have right now is this odd middleground of no exception, behaviour of the first, documentation of the second.

**Comment by iudeen** [comment]
I think flask raises Exception. I'll see if it's worthy of a PR. May be I can give a try.

**Comment by tiangolo** [comment]
Cool, yeah, I would take a PR with that, if @bbeattie-phxlabs or @iudeen want to take a stab at that it would be great! 🚀
